src/confd/src/ietf-system.c

```diff
@@ -1018,6 +1003,67 @@ static int change_auth(sr_session_ctx_t *session, uint32_t sub_id, const char *m
 	return SR_ERR_OK;
 }
 
+static int change_nacm(sr_session_ctx_t *session, uint32_t sub_id, const char *module,
+		       const char *xpath, sr_event_t event, unsigned request_id, void *priv)
+{
+	sr_val_t *users = NULL, *groups = NULL, *rules = NULL;
+	size_t user_count = 0, group_count = 0, rule_count = 0;
+	struct confd *confd = (struct confd *)priv;
+	int rc;
+
+	if (event != SR_EV_DONE)
+		return SR_ERR_OK;
+
+	/* Fetch all users from ietf-system */
+	rc = sr_get_items(session, XPATH_AUTH_"/user/name", 0, 0, &users, &user_count);
+	if (SR_ERR_OK != rc) {
+		ERROR("Failed fetching system users: %s", sr_strerror(rc));
+		goto cleanup;
+	}
+
+	aug_load(confd->aug);
+
+	for (size_t i = 0; i < user_count; i++) {
+		const char *user = users[i].data.string_val;
+		char xpath_groups[256];
+		bool is_admin = false;
+
+		/* Fetch groups for each user */
+		snprintf(xpath_groups, sizeof(xpath_groups), NACM_BASE_"/groups/group[user-name='%s']/name", user);
+		rc = sr_get_items(session, xpath_groups, 0, 0, &groups, &group_count);
+		if (!rc) {
+			for (size_t j = 0; j < group_count; j++) {
+				char xpath_rules[256];
+
+				/* Fetch and check rules for each group */
+				snprintf(xpath_rules, sizeof(xpath_rules), NACM_BASE_"/rule-list"
+					 "[group='%s']/rule[module-name='*'][access-operations='*']"
+					 "[action='permit']", groups[j].data.string_val);
```

Comment:
That's very nifty! :nerd_face:

Reply:
Sometimes the native sysrepo APIs really shine :nerd_face: